apps/dashboard/app/controllers/api/api_sessions_controller.rb

```diff
@@ -0,0 +1,142 @@
+class Api::ApiSessionsController < ApiController
+
+  def index
+    @sessions = BatchConnect::Session.all
+    @sessions.each(&:update_cache_completed!)
+
+    response = @sessions.map do |session|
+      create_session_data session
+    end
+
+    render json: { items: response }
+
+  rescue Exception => error
+    logger.error "action=getSessions user=#{@user} error=#{error}"
+    render json: { message: error }, status: :internal_server_error
+  end
+
+  def show
+    session_id = params[:id]
+    if !BatchConnect::Session.exist?(session_id)
+      render json: { message: "Not found session_id: #{session_id}" }, status: :not_found
+      return
+    end
+
+    session = BatchConnect::Session.find(session_id)
+    render json: create_session_data(session)
+  end
+
+  def create
+    if params[:token] == nil || params[:token] == ""
+      render json: { message: "missing token" }, status: :bad_request
+      return
+    end
+
+    app = BatchConnect::App.from_token params[:token]
+    if !app.valid?
+      render json: { message: app.validation_reason }, status: :bad_request
+      return
+    end
+
+    success, session = create_session
+    if success
+      render json: { id: session.id }
+    else
+      logger.error "action=createSession user=#{@user} errors=#{session.errors.full_messages}"
+      render json: { message: "Unable to create session", errors: session.errors.full_messages }, status: :internal_server_error
+    end
+
+  rescue Exception => error
+    logger.error "action=createSession user=#{@user} error=#{error}"
+    render json: { message: "Exception while creating session", errors: error }, status: :internal_server_error
+  end
+
+  def destroy
+    session_id = params[:id]
+    if !BatchConnect::Session.exist?(session_id)
+      render json: { message: "Not found sessionId: #{session_id}" }, status: :not_found
+      return
+    end
+
+    session = BatchConnect::Session.find(session_id)
+    if session.destroy
+      render json: {}, status: :no_content
+    else
+      logger.error "action=deleteSession user=#{@user} errors=#{session.errors.full_messages}"
+      render json: { message: "Unable to delete session", errors: session.errors.full_messages }, status: :internal_server_error
+    end
+
+  rescue Exception => error
+    logger.error "action=deleteSession user=#{@user} error=#{error}"
+    render json: { message: "Exception while deleting session", errors: error }, status: :internal_server_error
+  end
+
+  private
+
+  def create_session_data(session)
+    sessionDataUrl = OodAppkit.files.url(path: session.staged_root).to_s
+    sessionShellUrl = OodAppkit.shell.url(host: session.connect.host).to_s if session.running? && session.connect.host
+    connect = create_connect_data session
+
+    {
+      id: session.id,
+      clusterId: session.cluster_id,
+      jobId: session.job_id,
+      createdAt: session.created_at,
+      token: session.token,
+      title: session.title,
+      info: session.info.to_h,
```

Comment:
There could be sensitive data in `session.info` here (connection credentials). We need a way to sanitize it if we're returning things in an API.

Reply:
This is intentional. Clients of the API will need these credentials to create links/forms to connect to the applications launched with the API.

Another idea to avoid sharing these was to create a back-end service to connect to a particular session.
Like a link to: `/connect/<session-id>` will redirect the user to the application with an already created session.

But I do not think this is possible for all applications, so sharing the credentials seems like the only way.